front-end/src/Components/Fucntions.js

```diff
@@ -0,0 +1,116 @@
+export const onChangeHandler = (event, data, setData) => {
+    const temp = {};
+    temp[event.target.name] = event.target.value;
+
+    if (event.target.name === "Gender")
+        temp[event.target.name] = event.target.id;
+
+    if (data)
+        setData({ ...data, ...temp });
+    else
+        setData(temp);
+}
+
+export const onClickHandler = async (route, method, navigate, data) => {
+    console.log("in the on click handler for the "+method);
+    const response = await fetch(`/${route}`, {
+        method: method,
+        headers: {
+            'Content-Type': 'application/json',
+            'authorization': `Bearer ${localStorage.getItem("userToken")}`,
+        },
+        body: JSON.stringify({
+            userFirstName: data.FirstName,
+            userLastName: data.LastName,
+            userAge: data.Age,
+            userGender: data.Gender,
+            userProfession: data.Profession,
+        })
+    })
+
+    if (response.status === 200) {
+        console.log("navigate");
+        navigate('/Home');
+    }
+}
+
+export async function apiCall(navigate, setError, data) {
+    if (data['userName'] !== "" && data['userPassword'] !== "" && data['userEmail'] !== "") {
+
+        const response = await fetch("/registration", {
+            method: "POST",
+            headers: {
+                "CONTENT-TYPE": "application/json"
+            },
+            body: JSON.stringify({
+                userName: data['userName'], userPassword: data['userPassword'], userEmail: data['userEmail'],
+            })
+        });
+        console.log(response.status);
+        if (response.status === 200) {
+            setError(false);
+            navigate("/", { state: { email: data['userEmail'] } });
+        }
+        else {
+            setError(true);
+        }
+    }
+}
+
+export const buttonHandlerSignOut = (navigate) => {
+    console.log("buttonHandlerSignOut was called");
+    localStorage.removeItem("userToken");
```

Comment:
Very good that you used local Storage but kindly use local storage with extra care.

Reply:
okay I will take care in the future